deltakit-explorer/src/deltakit_explorer/analysis/_quops.py

```diff
@@ -0,0 +1,142 @@
+# (c) Copyright Riverlane 2020-2025.
+"""Module explores how the error suppression factor (lambda)
+builds a connection between QuOps and code distance.
+
+References
+----------
+- https://doi.org/10.48550/arXiv.2408.13687
+"""
+
+from typing import Callable
+
+
+def _equal_or_less_brute_force_search(
+    func: Callable[[int], float],
+    target: float,
+    minimum: int,
+    maximum: int,
+) -> int | None:
+    """Brute force search or the solution at on an interval.
+    
+    func(...) is (unfortunately) not always a monotonic function.
+    The function searches for smallest D from [min, max],
+    with func(D) <= target.
+
+    Parameters:
+        func (Callable): a descending function.
+        target (float): a value to search for.
+        minimum (int): minimum tested value of D.
+        maximum (int): maximum tested value of D.
+
+    Returns:
+        int | None:
+            a value, which safisfies the search, or None.
+    """
+    while minimum < maximum:
+        value = func(minimum)
+        if value <= target:
+            return minimum
+        minimum += 1
```

Comment:
Can be simplified (at least in my opinion that's a simplification) to:

```suggestion
    for test_input in range(minimum, maximum):
        value = func(test_input)
        if value <= target:
            return test_input
```